clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1098,8 +1098,20 @@ Node selectReadReplica(TopicPartition partition, Node leaderReplica, long curren
         Map<Node, FetchSessionHandler.Builder> fetchable = new LinkedHashMap<>();
 
         // Ensure the position has an up-to-date leader
-        subscriptions.assignedPartitions().forEach(
-            tp -> subscriptions.maybeValidatePositionForCurrentLeader(tp, metadata.currentLeader(tp)));
+        subscriptions.assignedPartitions().forEach(tp -> {
+            Metadata.LeaderAndEpoch leaderAndEpoch = metadata.currentLeader(tp);
+            if (leaderAndEpoch.leader.isPresent()) {
+                NodeApiVersions nodeApiVersions = apiVersions.get(leaderAndEpoch.leader.get().idString());
```

Comment:
Would it make sense to push this check into `maybeValidatePositionForCurrentLeader` by passing through `ApiVersions`?  Otherwise, some of the checks here are redundant.

Reply:
If we do this, do you think we should automatically call `completeValidation` from `maybeValidatePositionForCurrentLeader`?

Reply:
Yes, that makes sense to me.